src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1389,79 +1389,127 @@ def get_network_lb(cli_ctx, resource_group_name, lb_name):
             # namespace.use_unmanaged_disk,
-            namespace.vm_domain_name,
-            namespace.vm_sku,
+            # namespace.vm_domain_name,
+            # namespace.vm_sku,
             # namespace.vnet_address_prefix,
-            namespace.vnet_name
+            # namespace.vnet_name
         ]
         if any(param is not None for param in banned_params):
             raise CLIError('usage error: In VM mode, only name, resource-group, location, '
                            'tags, zones, platform-fault-domain-count, single-placement-group and ppg are allowed')
+
+        # if namespace.platform_fault_domain_count is None:
+        #     raise CLIError("usage error: --platform-fault-domain-count is required in Flexible mode")
+
+        if namespace.tags is not None:
+            validate_tags(namespace)
+        _validate_location(cmd, namespace, namespace.zones, namespace.vm_sku)
+        # validate_edge_zone(cmd, namespace)
+        if namespace.application_security_groups is not None:
+            validate_asg_names_or_ids(cmd, namespace)
+        if getattr(namespace, 'attach_os_disk', None) or namespace.image is not None:
+            _validate_vm_create_storage_profile(cmd, namespace, for_scale_set=True)
+        if namespace.vnet_name or namespace.subnet:
+            _validate_vm_vmss_create_vnet(cmd, namespace, for_scale_set=True)
+
+        if namespace.load_balancer is not None or namespace.application_gateway is not None:
+            _validate_vmss_create_load_balancer_or_app_gateway(cmd, namespace)
+        if namespace.public_ip_address is not None:
+            _validate_vmss_create_public_ip(cmd, namespace)
+        if namespace.nsg is not None:
+            _validate_vmss_create_nsg(cmd, namespace)
+        if namespace.accelerated_networking is not None:
+            _validate_vm_vmss_accelerated_networking(cmd.cli_ctx, namespace)
+        if namespace.admin_password or namespace.ssh_dest_key_path or getattr(namespace, 'ssh_dest_key_value', None):
+            _validate_vm_vmss_create_auth(namespace, cmd)
+        if namespace.assign_identity == '[system]':
+            raise InvalidArgumentValueError('usage error: only user assigned indetity is suppoprted for Flex mode.')
+        if namespace.assign_identity is not None:
+            _validate_vm_vmss_msi(cmd, namespace)  # -- UserAssignedOnly
         _validate_proximity_placement_group(cmd, namespace)
+        _validate_vmss_terminate_notification(cmd, namespace)
+        if namespace.automatic_repairs_grace_period is not None:
+            _validate_vmss_create_automatic_repairs(cmd, namespace)
+        _validate_vmss_create_host_group(cmd, namespace)
+
+        if namespace.secrets is not None:
+            _validate_secrets(namespace.secrets, namespace.os_type)
+
+        if namespace.eviction_policy and not namespace.priority:
+            raise ArgumentUsageError('usage error: --priority PRIORITY [--eviction-policy POLICY]')
```

Comment:
There is only a few `if` in uniform mode. Can we write only one copy of code? Most of these lines are duplicate code compared with uniform mode.

Reply:
as in flex mode, we could not create resources like uniform mode, but we want to utilize the check/validation, too